tests/test_bytecode_circuit.py

```diff
@@ -85,7 +87,7 @@ def test_bytecode_multiple():
         unroll(bytes([Opcode.ADD, Opcode.PUSH32]), randomness),
         unroll(bytes([Opcode.ADD, Opcode.PUSH32, Opcode.ADD]), randomness),
     ]
-    verify(k, bytecodes, randomness, True)
+    verify(k, bytecodes, randomness, False)  # Push without data must fail
```

Comment:
Should it? I thought this was currently allowed but I could be wrong.

Reply:
Is it allowed? Is this because we should allow invalid bytecode to run in the EVM as it would? The EVM would rise an exception and revert the transaction?

In that case perhaps we need some changes to the spec.

Reply:
As far as I know this currently is allowed, future EIPs like EIP-3670 may introduce these kinds of checks but I don't think any checks (except on total bytecode size) are currently being done.

Executing this kind of code will result in a revert (which is totally valid to do), but the bytecode of a smart contract may also just just contain sections that are never used by transactions but still needs to be loaded in.

Reply:
I think push without data should be allowed and it's will be padded with `n` zero when EVM really executes this `PUSHn`.

The reason this case fails here is because there will be `next_push_data_left` unassigned in the padding rows, we might need to assign it for the first padding row to pass the `assert next.push_data_left == cur.push_data_size`

Reply:
@han0110 @Brechtpd what I have done (that it seemed more semantic to me but perhaps there is something wrong) is to only assert `assert next.push_data_left ==` when we are in a Byte -> Byte transition, please let me know if there is something wrong with this approach.

Reply:
This approach also looks much better to me!